packages/allow-scripts/src/index.js

```diff
@@ -83,141 +244,242 @@ async function runScript ({ path, event }) {
   }
 
   if (missingPolicies.length) {
-    console.log('\n# unconfigured packages!')
+    console.log('\n# unconfigured packages with lifecycle scripts')
     missingPolicies.forEach(pattern => {
-      const collection = packagesWithLifecycleScripts.get(pattern) || []
+      const collection = packagesWithScripts.get(pattern) || []
       console.log(`- ${pattern} [${collection.length} location(s)]`)
     })
   }
 
   if (excessPolicies.length) {
-    console.log('\n# packages that dont need configuration (missing or no lifecycle scripts)')
+    console.log('\n# packages with lifecycle scripts that no longer need configuration due to package or scripts removal')
     excessPolicies.forEach(pattern => {
-      const collection = packagesWithLifecycleScripts.get(pattern) || []
+      const collection = packagesWithScripts.get(pattern) || []
       console.log(`- ${pattern} [${collection.length} location(s)]`)
     })
   }
 }
 
-async function loadAllPackageConfigurations ({ rootDir }) {
-  const packagesWithLifecycleScripts = new Map()
+/**
+ * 
+ * @param {Object} args 
+ * @param {string} args.rootDir
+ * @param {PkgConfs} args.conf
+ * @returns {Promise}
+ */
+async function savePackageConfigurations({ rootDir, conf: {
+  packageJson,
+  configs: { lifecycle, bin }
+} }) {
+  // update package json
+  if (!packageJson.lavamoat) packageJson.lavamoat = {}
+  packageJson.lavamoat.allowScripts = lifecycle.allowConfig
+  packageJson.lavamoat.allowBins = bin.allowConfig
+  const packageJsonPath = path.resolve(rootDir, 'package.json')
+  const packageJsonSerialized = JSON.stringify(packageJson, null, 2) + '\n'
+  await fs.writeFile(packageJsonPath, packageJsonSerialized)
+}
+
+/**
+ * 
+ * @param {Object} args 
+ * @param {string} args.rootDir
+ * @returns {Promise<PkgConfs>}
+ */
+async function loadAllPackageConfigurations({ rootDir }) {
+  const packagesWithScriptsLifecycle = new Map()
+  const binCandidates = new Map()
 
   const dependencyMap = await loadCanonicalNameMap({ rootDir, includeDevDeps: true })
   const sortedDepEntries = Array.from(dependencyMap.entries()).sort(sortBy(([filePath, canonicalName]) => canonicalName))
+  const packageJson = JSON.parse(await fs.readFile(path.join(rootDir, 'package.json'), 'utf8'))
+  const directDeps = new Set([...Object.keys(packageJson.devDependencies||{}),...Object.keys(packageJson.dependencies||{})])
+
   for (const [filePath, canonicalName] of sortedDepEntries) {
     // const canonicalName = getCanonicalNameForPath({ rootDir, filePath: filePath })
     let depPackageJson
     try {
-      depPackageJson = JSON.parse(await fs.readFile(path.join(filePath, 'package.json')))
+      depPackageJson = JSON.parse(await fs.readFile(path.join(filePath, 'package.json'), 'utf-8'))
     } catch (err) {
-      const branchIsOptional = branch.some(node => node.optional)
-      if (err.code === 'ENOENT' && branchIsOptional) {
-        continue
-      }
+      // FIXME: leftovers of code that used to work
```

Comment:
Fix or remove..?

Reply:
This was broken for a long time. I didn't want to fix it in this feature.
Left it as an eyesore so that I'm more likely to investigate what feature this was supporting many refactorings ago. Looking at this is meant to hurt :smiling_imp: